components/script/dom/servoparser/mod.rs

```diff
@@ -175,6 +175,7 @@ impl ServoParser {
         document.set_current_parser(Some(&parser));
         if !type_.eq_ignore_ascii_case("text/html") {
             parser.parse_string_chunk("<pre>\n".to_owned());
+            assert!(parser.tokenizer.try_borrow_mut().is_ok());
```

Comment:
Asserting here is no different than performing the infallible borrow. Rather than adding these extra asserts everywhere, I would like a single API on ServoParser called something like `parser_is_not_active` (and an API on Document named something like `can_invoke_script`, which has access to the parser), and we should assert that that API returns true when we're asked to dispatch an event in EventTarget.